userspace/libsinsp/plugin.cpp

```diff
@@ -449,13 +453,35 @@ std::list<sinsp_plugin::info> sinsp_plugin::plugin_infos(sinsp* inspector)
 	return ret;
 }
 
-sinsp_plugin::sinsp_plugin()
-	: m_nfields(0)
+bool sinsp_plugin::is_plugin_loaded(sinsp* inspector, std::string &filepath)
+{
+#ifdef _WIN32
+	// This returns an HMODULE indeed, but they are the same thing
+	sinsp_plugin_handle handle = (HINSTANCE)GetModuleHandle(filepath.c_str());
+#else
+	sinsp_plugin_handle handle = dlopen(filepath.c_str(), RTLD_LAZY | RTLD_NOLOAD);
```

Comment:
I didn't know that dlopen returns the same handle if you dlopen() a shared library a second time. But when I read the man page for dlopen(), I see:

> If the same library is loaded again with dlopen(), the same file handle is returned. The dl library maintains reference counts for library handles, so a dynamic library is not deallocated until dlclose() has been called on it as many times as dlopen() has succeeded on it. 

So wouldn't you need to dlclose() it here?

Reply:
Nope, as `RTLD_NOLOAD` does not increment reference count on the dlhandle:
> Don't load the library. This can be used to test if the library is already resident (dlopen() returns NULL if it is not, or the library's handle if it is resident).

Reply:
Got it, thanks!